packages/components/bucket-policy/bucket-policy-body.tsx

```diff
@@ -0,0 +1,442 @@
+        [
+          <Button
+            key="data-source-add-modal"
+            className="create-data-source__modal-button"
+            variant="plain"
+            onClick={() => {
+              setOpen(false);
+              launchModal();
+            }}
+          >
+            {t('Add new data source')}
+          </Button>,
+          <Divider component="li" key={'data-source-divider'} />,
+        ]
+      ),
+    [t, data, disableAllExcept, launchModal]
+  );
+
+  return (
+    <Select
+      {...props}
+      className={classNames(props?.className)}
+      variant={variant}
+      aria-label={t('Select data source')}
+      onToggle={setOpen}
+      onSelect={(e, selection) => {
+        onChange(selection as string);
+        variant !== SelectVariant.checkbox && setOpen(false);
+      }}
+      isOpen={isOpen}
+      id="data-source-dropdown"
+      placeholderText={t('Select options')}
+      noResultsFoundText={t('No results found')}
+    >
+      {dropdownOptions}
+    </Select>
+  );
+};
+
+const SingleDataResource: React.FC<DataResourceProps> = ({
+  state,
+  dispatch,
+  launchModal,
+  ...props
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <FormGroup
+      fieldId="read-write-data-source"
+      className="data-source-type__form--margin"
+      label={t('Select a read and write data source')}
+    >
+      <DataResourceDropdown
+        {...props}
+        selections={state.readWriteSingle}
+        className="create-bucket-policy__form--margin-bottom"
+        launchModal={() => launchModal(EDIT_DATA_RESOURCES, null)}
+        onChange={(selection) =>
+          dispatch({
+            type: BucketPolicyActionType.SET_READ_WRITE_SINGLE,
+            payload: selection,
+          })
+        }
+      />
+      <Checkbox
+        id="cache-checkbox"
+        label={t('Enable cache')}
+        onChange={(checked) =>
+          dispatch({
+            type: BucketPolicyActionType.SET_CACHE,
+            payload: checked,
+          })
+        }
+        isChecked={state.cacheEnabled}
+        description={t(
+          'The caching bucket will serve data from a large raw data out of a local caching tiering'
+        )}
+      />
+    </FormGroup>
+  );
+};
+
+const MultiDataResource: React.FC<DataResourceProps> = ({
+  state,
+  dispatch,
+  launchModal,
+  ...props
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <div className="data-source-type__form--margin">
+      <FormGroup fieldId="read-data-source" label={t('Read data source')}>
+        <p className="pf-c-form__helper-text">
+          {t(
+            'Select the data sources that defines the read targets of the bucket policy'
+          )}
+        </p>
+        <DataResourceDropdown
+          {...props}
+          variant={SelectVariant.checkbox}
+          selections={state.readResourceMulti}
+          className="create-bucket-policy__form--margin-bottom"
+          launchModal={() => launchModal(EDIT_DATA_RESOURCES, null)}
```

Comment:
instead of passing null, make extraProps optional

Reply:
ack...